### PR TITLE
Enable PAL with expansion audio

### PR DESCRIPTION
**Pull request by GreatGlobox**
There are a lot of changes, so I felt it was best to create a separate branch on my end for this and submit a separate PR from the volume mixing one. (296 of the changes are just new generated NSF and ROM files, and 50 are notetable.bin files, so it's not as crazy as it looks)

Here's a brief summary of what has been changed:

- Disabled multiple assert checks which don't allow PAL with expansion audio
- Enabled several UI options that would normally be disabled when using expansions
- Added PAL pitch tables, and also added them to the assembly so they will be used for export
- Enabled exporting PAL NSF and ROM files with expansion audio, and generated all the necessary files
- Made changes to the documentation and tooltips to remove mentions of no PAL + expansion audio
- Made changes to NotSoFatso so that it would import PAL expansion audio
- Optimisations for assembly by moving note tables to separate .bin files
- ~Fixed NESASM3 and asm6 so they search relative paths to nested includes~

### CODE REVIEW

**Comment by BleuBleu** [comment]
This one is up to date and looks good.

Minor fixes/nitpicks:
- Why did we change the default bass cutoff (16 -> 24) in the settings? Is it part of there square/TND change that got mixed up in this PR?
- Missing MIDI file PAL import. Please do a quick review at all import/export format to see if changes are needed, I may have missed others. Does VGM need anything for example?
- In "NesApu.cs", some weird `using` appeared, can you removed those if they arent needed? They will likely not compile on Android, etc.

**Comment by GreatGlobox** [comment]
I changed it from 16 to 24 because it was matching the original hardware (Famicom) better in tests. It's possible that it did get mixed up with this PR as it's supposed to be volume related. My mistake on that part if so.

I will double check those things in terms of export, my bad on this. I should go through it all and make sure everything is functional.

VS Code and Visual Studio keep doing this and adding random "using" things that are not even needed. I have had to go back and remove those a few times. It kept doing this with SharpDX and breaking Linux, and that VisualBasic one is likely another case of this. We can just remove them.

**Comment by GreatGlobox** [comment]
For MIDI, there is a "Create PAL Project" checkbox already, which imports it and creates a PAL project. I will have to double check the other things like VGM though. I will look into it further later today.

**Comment by BleuBleu** [comment]
Thanks, yeah for MIDI that was it, just that checkbox that was doing the "old" behavior. And yes, regular VS also adds random `using` all the time.

**Comment by GreatGlobox** [comment]
Yeah that makes sense then, now it seems to work.

For VGM, I just tested it and it seems that if you export a VGM from a PAL song, then import it again, the project ends up as NTSC but at the same tempo as PAL (the correct tempo), while using NTSC note tables (higher pitched). This still happens even if you're not using any expansions and I assume this is not expected behaviour. Purposefully setting pal to true in VS Code makes the pitch correct, but the song gets even slower than it should be.

In short; VGM import from PAL is the correct tempo, but wrong machine type and using the wrong notetables, regardless of expansion audio.

**Comment by BleuBleu** [comment]
Is this one ready for final-final review or your are still working on it?

**Comment by GreatGlobox** [comment]
It should be good now. 2 changes were only to make Linux debugging work and not keep stuttering so I could figure out VGM. It just flat out wouldn't work in debug mode unless GL callback was synchronous for some reason.

**Comment by BleuBleu** [comment]
Ok, one very last question: Why are we using a different ASM6, why did we comment out the one we use for unit tests?

**Comment by GreatGlobox** [comment]
This was both asm6 and nesasm3. They couldn't find the paths during unit tests because they didn't find the path relative to an included file, so I had made forks of them and changed how they find the paths. They were looking in the working directory for the SoundEngine/NoteTables directory instead of up one level like we wanted (the current included file). It should now search the included file directory first and then fall back to the working directory. Our idea of using bin files wasn't working just because they couldn't find paths.

**Comment by BleuBleu** [comment]
I dont want weird forks of common tools people use. 

If it doesnt find the includes, we should figure out a way to make it work with the "stock" tools. We could remove the \NoteTables subfolder, for example.

-Mat

**Comment by GreatGlobox** [comment]
No problem. I think this means needing to have the notetables actually be present in the same folder as the unit tests themselves. It still works if they're in a subfolder, but what doesn't work is the fact they're up one level in the SoundEngine folder instead of UnitTests. It's looking in the same directory we ran the bat file from. Unless there is a way to make the bat file change directory first without it breaking anything.

**Comment by GreatGlobox** [comment]
Doing this seems to make the tests work with the stock tools (ca65 works, the directory only needs to change for the other assembly tools):

`:: Compile everything.
..\..\Tools\ca65 test_ca65.s -g -o test_ca65.o
..\..\Tools\ld65 -C test_ca65.cfg -o test_ca65.nes test_ca65.o --mapfile test_ca65.map --dbgfile test_ca65.dbg
pushd ..
REM ..\Tools\asm6 UnitTests\test_asm6.asm UnitTests\test_asm6.nes
..\Tools\asm6_fixed UnitTests\test_asm6.asm UnitTests\test_asm6.nes
..\Tools\NESASM3 UnitTests\test_nesasm.asm
popd`

You will notice I have it using asm6 instead of asm6_fixed though. The tests are failing with asm6_fixed for some reason. No idea why.
